### PR TITLE
Adding waitForAllDatabasesStatusUnderSubscription function

### PR DESCRIPTION
**Pull request by Linoyperetz**

### CODE REVIEW

File: api.ts [hunk 1]
```diff
@@ -588,6 +588,26 @@ export class CloudAPISDK {
...
         return database;
     }
+    /**
+     * Waiting for all databases status under subscription to change to 'Active'
+     * @param subscriptionId The id of the subscription
+     * @param expectedStatus The expected status
+     * @param timeoutInSeconds The timeout of waiting for the status. Default: 5 minutes
+     * @param sleepTimeInSeconds The sleep time between requests. Default: 5 seconds
+     */
+    async waitForAllDatabasesStatusUnderSubscription(subscriptionId: number, timeoutInSeconds = 5 * 60, sleepTimeInSeconds = 5) {
```
Comment (danitseitlin) [inline api.ts:598]: Please rename to `waitForSubscriptionDatabasesStatus`

Comment (danitseitlin) [inline api.ts:598]: Please add a parameter for status while active is default value

File: api.ts [hunk 2]
```diff
@@ -588,6 +588,26 @@ export class CloudAPISDK {
...
+        let databases = await this.getDatabases(subscriptionId);
+        let timePassedInSeconds = 0;
+        for ( let database of databases){
+            while (database.status !== 'active' && database.status !== 'error' && database.status !== undefined && timePassedInSeconds <= timeoutInSeconds) { 
+                this.log('debug', `Waiting for database ${database.databaseId} status '${database.status}' to be become status 'active' (${timePassedInSeconds}/${timeoutInSeconds} (Subscription ${subscriptionId})`);
+                await this.sleep(sleepTimeInSeconds);
+                timePassedInSeconds+=sleepTimeInSeconds;
+                database = await this.getDatabase(subscriptionId, database.databaseId);
+            }
+            this.log('debug', `Database ${database.databaseId} ended up as '${database.status}' status after ${timePassedInSeconds}/${timeoutInSeconds} (Subscription ${subscriptionId})`)
```
Comment (danitseitlin) [inline api.ts:608]: Why not use waitForDatabaseStatus here instead?

File: api.ts [hunk 3]
```diff
@@ -588,6 +588,26 @@ export class CloudAPISDK {
...
+     * Waiting for all databases status under subscription to change to 'Active'
+     * @param subscriptionId The id of the subscription
+     * @param expectedStatus The expected status
+     * @param timeoutInSeconds The timeout of waiting for the status. Default: 5 minutes
+     * @param sleepTimeInSeconds The sleep time between requests. Default: 5 seconds
+     */
+    async waitForAllDatabasesStatusUnderSubscription(subscriptionId: number, timeoutInSeconds = 5 * 60, sleepTimeInSeconds = 5) {
+        let databases = await this.getDatabases(subscriptionId);
+        let timePassedInSeconds = 0;
+        for ( let database of databases){
```
Comment (danitseitlin) [inline api.ts:601]: please remove redundant spaces

File: api.ts [hunk 4]
```diff
@@ -588,6 +588,26 @@ export class CloudAPISDK {
         this.log('debug', `Database ${databaseId} ended up as '${database.status}' status after ${timePassedInSeconds}/${timeoutInSeconds} (Subscription ${subscriptionId})`);
         return database;
     }
+    /**
+     * Waiting for all databases status under subscription to change to 'Active'
```
Comment (danitseitlin) [inline api.ts:592]: to expected status

**Review by danitseitlin — CHANGES_REQUESTED** [review]

File: api.ts (hunk 2 above)
Comment (Linoyperetz) [reply api.ts:608]: You're very right

File: api.ts [hunk 5]
```diff
@@ -588,7 +588,20 @@ export class CloudAPISDK {
...
+     * Waiting for all databases status under subscription to change to the expected status
+     * @param subscriptionId The id of the subscription
+     * @param expectedStatus The expected status
+     * @param timeoutInSeconds The timeout of waiting for the status. Default: 5 minutes
+     * @param sleepTimeInSeconds The sleep time between requests. Default: 5 seconds
+     */
+    async waitForSubscriptionDatabasesStatus(subscriptionId: number, expectedStatus: DatabaseStatus = 'active', timeoutInSeconds = 5 * 60, sleepTimeInSeconds = 5) {
+        let databases = await this.getDatabases(subscriptionId);
+        for (let database of databases){
+            this.waitForDatabaseStatus(subscriptionId, database.databaseId, expectedStatus, timeoutInSeconds, sleepTimeInSeconds)
```
Comment (danitseitlin) [inline api.ts:601]: Where is your await? 🤔

File: api.ts [hunk 6]
```diff
@@ -588,7 +588,20 @@ export class CloudAPISDK {
...
+    /**
+     * Waiting for all databases status under subscription to change to the expected status
+     * @param subscriptionId The id of the subscription
+     * @param expectedStatus The expected status
+     * @param timeoutInSeconds The timeout of waiting for the status. Default: 5 minutes
+     * @param sleepTimeInSeconds The sleep time between requests. Default: 5 seconds
+     */
+    async waitForSubscriptionDatabasesStatus(subscriptionId: number, expectedStatus: DatabaseStatus = 'active', timeoutInSeconds = 5 * 60, sleepTimeInSeconds = 5) {
+        let databases = await this.getDatabases(subscriptionId);
+        for (let database of databases){
```
Comment (danitseitlin) [inline api.ts:600]: i'm pretty sure this could be a const 😂

File: api.ts [hunk 7]
```diff
@@ -588,7 +588,20 @@ export class CloudAPISDK {
         this.log('debug', `Database ${databaseId} ended up as '${database.status}' status after ${timePassedInSeconds}/${timeoutInSeconds} (Subscription ${subscriptionId})`);
         return database;
     }
-
```
Comment (danitseitlin) [inline api.ts:591]: I'd add this back, I love the spacing between functions. maybe it more split and readable

**Review by danitseitlin — CHANGES_REQUESTED** [review]

File: api.ts (hunk 7 above)
Comment (Linoyperetz) [reply api.ts:591]: it splitted, instead of 2 lines space, 1 line now

**Review by danitseitlin — APPROVED** [review]

**Pull request merged**